src/Headers.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Minishlink\WebPush;
+
+use ArrayIterator;
+use InvalidArgumentException;
+use IteratorAggregate;
+use Traversable;
+
+class Headers implements IteratorAggregate
+{
+    /**
+     * @var array
+     */
+    private $headers;
+
+    public function __construct(array $headers = [])
+    {
+        $this->headers = $headers;
+    }
+
+    /**
+     * @param Headers|array $headers
+     *
+     * @return Headers
+     */
+    public static function wrap($headers): self
+    {
+        return $headers instanceof self ? $headers : new static((array) $headers);
+    }
+
+    /**
+     * @param Headers|array $headers
+     *
+     * @return Headers
+     */
+    public function with($headers): Headers
+    {
+        return new self(array_replace(self::wrap($headers)->toArray(), $this->headers));
+    }
+
+    /**
+     * @param string $name
+     * @param string|array|int $value
+     *
+     * @return Headers
+     * @throws InvalidArgumentException
+     */
+    public function set(string $name, $value): self
+    {
+        if (is_scalar($value) === false && is_array($value) === false) {
+            throw new InvalidArgumentException(sprintf('Invalid value specified for %s header', $name));
+        }
+
+        $this->headers[$name] = $value;
+
+        return $this;
+    }
+
+    /**
+     * @param string $name
+     *
+     * @return mixed Returns null if a header does not exist.
+     */
+    public function get(string $name)
+    {
+        return $this->headers[$name] ?? null;
```

Comment:
`null` may be a valid value.
From my POV it is preferable to throw an exception if `$name` does not exist.
It supposes to check with `has` before

Reply:
> `null` may be a valid value.
> From my POV it is preferable to throw an exception if `$name` does not exist.
> It supposes to check with `has` before

That sounds fair, willing to make that change for sure 👍

Reply:
Addressed in e96d01e